middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -1578,6 +1482,244 @@ let close_let_rec acc env ~function_declarations
       named ~body
     |> Expr_with_acc.create_let
 
+let wrap_partial_application acc env apply_continuation (apply : IR.apply)
+    approx args missing_args ~arity ~num_trailing_local_params
+    ~contains_no_escaping_local_allocs =
+  (* In case of partial application, creates a wrapping function from scratch to
+     allow inlining and lifting *)
+  let wrapper_id = Ident.create_local ("partial_" ^ Ident.name apply.func) in
+  let closure_id =
+    Closure_id.wrap
+      (Compilation_unit.get_current_exn ())
+      (Variable.create_with_same_name_as_ident wrapper_id)
+  in
+  let args_arity = List.length args in
+  let params =
+    List.mapi
+      (fun n _ ->
+        ( Ident.create_local ("param" ^ string_of_int (args_arity + n)),
+          Lambda.Pgenval ))
```

Comment:
I'm a bit disappointed that we're not recovering a better kind from the arity of the function. That would require to convert from Flambda kinds back into Lambda kinds, and I don't think we have the code for that yet, but I think we should at least leave a CR about it.